3.analyze_data/utils/significance_testing.py

```diff
@@ -134,37 +134,52 @@ def get_treatment_comparison(_comp_functions, _treatment_paths, _probadf, _barco
+        # Create groups for making comparisons using the model type and the treatment
+        # We only make comparisons within the same phenotype, which is performed below
+        iter_group = set(zip(no_negcondf["Metadata_model_type"], no_negcondf[treat_data["treatment_column"]]))
 
         # Iterate through each group
-        for plate, model_type, utreat in iter_group:
-
-            # Specify the treatment and negative control dataframes
-            treatdf = no_negcondf.loc[(no_negcondf["Metadata_plate"] == plate) & (no_negcondf["Metadata_model_type"] == model_type) & (no_negcondf[treat_data["treatment_column"]] == utreat)]
-            negdf = negcondf.loc[(negcondf["Metadata_plate"] == plate) & (negcondf["Metadata_model_type"] == model_type)]
-
-            # Remove wells if the cell count is below the corresponding threshold
-            treatdf = filter_wells_by_cell_count(treatdf, _treat_cutoff)
-            negdf = filter_wells_by_cell_count(negdf, _control_cutoff)
-
-            # Compute the number of cells for each group
-            treat_cell_count = len(treatdf)
-            negcon_cell_count = len(negdf)
-            min_cell_count = min(treat_cell_count, negcon_cell_count)
-
-            # If there are no probability values that match the given well for some reason analyze the next treatment
-            if (min_cell_count > 0):
-
-                # Sample the treatment dataframe if the cell count for the treatments is larger than for the controls
-                if treat_cell_count > negcon_cell_count:
-                    samp_treat = treatdf.sample(n=min_cell_count, random_state=0)
-                    samp_neg = negdf
-
-                # Otherwise, sample the negative control
-                else:
-                    samp_neg = negdf.sample(n=min_cell_count, random_state=0)
-                    samp_treat = treatdf
+        for model_type, utreat in iter_group:
+            samp_treat = []
+            samp_neg = []
+            is_empty = True
+
+            groupdf = no_negcondf.loc[(no_negcondf["Metadata_model_type"] == model_type) & (no_negcondf[treat_data["treatment_column"]] == utreat)]
+
+            # Compute the comparisons within each plate seperately
+            for plate in groupdf["Metadata_plate"].unique():
+
+                # Specify the treatment and negative control dataframes
+                treatdf = no_negcondf.loc[(no_negcondf["Metadata_plate"] == plate) & (no_negcondf["Metadata_model_type"] == model_type) & (no_negcondf[treat_data["treatment_column"]] == utreat)]
+                negdf = negcondf.loc[(negcondf["Metadata_plate"] == plate) & (negcondf["Metadata_model_type"] == model_type)]
+
+                # Remove wells if the cell count is below the corresponding threshold
+                treatdf = filter_wells_by_cell_count(treatdf, _treat_cutoff)
```

Comment:
I am curious as to how these cutoffs are defined?

Reply:
I just choose a constant to define the cutoffs.